al2022arm.pkr.hcl

```diff
@@ -0,0 +1,33 @@
+locals {
+  ami_name_al2022arm = "${var.ami_name_prefix_al2022}-hvm-2022.0.${var.ami_version}-arm64-ebs"
+}
+
+source "amazon-ebs" "al2022arm" {
+  ami_name        = "${local.ami_name_al2022arm}"
+  ami_description = "Amazon Linux AMI 2022.0.${var.ami_version} arm64 ECS HVM GP2"
+  instance_type   = "m6g.xlarge"
+  launch_block_device_mappings {
+    volume_size           = var.block_device_size_gb
+    delete_on_termination = true
+    volume_type           = "gp2"
+    device_name           = "/dev/xvda"
+  }
+  region = var.region
+  source_ami_filter {
+    filters = {
+      name = "${var.source_ami_al2022arm}"
+    }
+    owners      = ["amazon"]
+    most_recent = true
+  }
+  user_data_file = "scripts/al2022/user-data.sh"
+  ssh_username   = "ec2-user"
+  tags = {
+    os_version          = "Amazon Linux 2022"
+    source_image_name   = "{{ .SourceAMIName }}"
+    ecs_runtime_version = "Docker version ${var.docker_version}"
+    ecs_agent_version   = "${var.ecs_agent_version}"
+    ami_type            = "al2022arm"
+    ami_version         = "2022.0.${var.ami_version}"
+  }
+}
```

Comment:
arm is WIP, correct?  That's why we don't have everything from [al2022.pkr.hcl](https://github.com/aws/amazon-ecs-ami/pull/43/files#diff-0acd7a3e28ca2dbe8c407c940343c3ff2ec92452343fa88cc37cf039fb2a32b8)?

Reply:
Although AL2022 ARM is not included in preview, due to https://github.com/amazonlinux/amazon-linux-2022/issues/101, but we have the HCL template [al2022arm.pkr.hcl](https://github.com/chienhanlin/amazon-ecs-ami/blob/mergeAL2022ToMain/al2022arm.pkr.hcl) ready for AL2022 ARM.